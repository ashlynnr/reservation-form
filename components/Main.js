import React, { useState } from "react";
import styled from "styled-components";
import rooms from "../utils/data";
import Form from "./Form";

const Main = () => {
  const [counter, setCounter] = useState(0);
  const [order, setOrder] = useState([{ id: 1, adults: 1, children: 0 }]);

  const orderHandler = (id, value) => {
    order.map(data => {
      if (data.hasOwnProperty(id)) {
        console.log(data.id);
      }
      console.log(id, value);
    });

    setOrder();
  };
  return (
    <Container>
      <Wrapper>
        {rooms.map((room, i) => (
          <Form
            key={room.id}
            room={room}
            checked={i <= counter}
            onCheck={() => setCounter(i)}
            order={order}
            orderHandler={orderHandler}
          />
        ))}
      </Wrapper>
      <Button>Submit</Button>
    </Container>
  );
};

export default Main;

const Container = styled.div`
  box-sizing: border-box;
  display: grid;
  grid-template-rows: repeat(auto-fill, minmax(500px, 90% 10%));
  height: 50vh;
  width: 100%;
  padding: 20px;
`;

const Button = styled.button`
  height: 5vh;
  width: 6vw;
`;

const Wrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(200px, 5fr));
  grid-gap: 20px;
  height: 150px;
  width: 100%;
`;
