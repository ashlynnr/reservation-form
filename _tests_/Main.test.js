import React from "react";
import { shallow } from "enzyme";
import Main from "../components/Main";
import configureStore from "redux-mock-store";

const initialState = { order: [{ id: 1, adults: 1, children: 0 }] };
const mockStore = configureStore();
let store, container;

beforeEach(() => {
  store = mockStore(initialState);
  container = shallow(<Main store={store} />);
});

it("Component Renders", () => {
  expect(container.length).toEqual(1);
});

it("Prop matches with initialState", () => {
  expect(container.prop("output")).toEqual(initialState.output);
});

describe("Main", () => {
  it("matches snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
