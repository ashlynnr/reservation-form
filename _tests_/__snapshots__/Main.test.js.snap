// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches snapshot 1`] = `
<ContextProvider
  value={null}
>
  <Main
    getOrder={[Function]}
    order={
      Array [
        Object {
          "adults": 1,
          "children": 0,
          "id": 1,
        },
      ]
    }
    reviseOrder={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
